File: gettls/week24/124 나라의 숫자.cpp
#include <string>
#include <vector>
#include <cmath>
using namespace std;

string num = "124";

string solution(int n) {
    string answer = "";
    int tmp = n;

    while (tmp > 0) {
        int x;
        if (tmp % 3 == 0) {
            x = 3;
            tmp = tmp / 3 - 1;
        }
        else {
            x = tmp % 3;
            tmp /= 3;
        }
        answer = num[x - 1] + answer;
    }
    return answer;
}

File: wonseok970/Programmers/124 나라의 숫자.py
def solution(n):
    answer = ''
    result = ''
    while n:
        if n % 3 == 0:
            answer += '4'
            n = n//3 - 1
        else:
            answer += str(n % 3)
            n = n//3

    for i in range(len(answer)-1, -1, -1):
        result += answer[i]

    return result

print(solution(10))